src/olympia/users/tests/test_tasks.py

```diff
@@ -172,3 +178,138 @@ def test_resize_photo_poorly():
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            callback=timeout_callback
+        )
+
+        with raises(Timeout):
+            sync_blocked_emails()
+
+        # Asser that task is retried
+
+    def test_empty_csv(self):
+        csv = list_to_csv([])
+
+        responses.add(
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        assert EmailBlock.objects.count() == 0
+
+    def test_existing_email(self):
+        user = user_factory()
+
+        csv = list_to_csv([user.email])
+
+        responses.add(
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            body=csv,
+            status=200,
+        )
+
+        EmailBlock.objects.create(email=user.email)
+
+        sync_blocked_emails()
+
+        email_block = EmailBlock.objects.get(email=user.email)
+
+        assert email_block.email == user.email
+        assert EmailBlock.objects.count() == 1
+
+    def test_unique_email(self):
+        user = user_factory()
+
+        csv = list_to_csv([user.email])
+
+        responses.add(
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        email_block = EmailBlock.objects.get(email=user.email)
+
+        assert email_block.email == user.email
+        assert EmailBlock.objects.count() == 1
+
+    def test_delete_file_if_error_processing_records(self):
+        csv = list_to_csv([True])
+
+        # mock uuid.uuid4 method to return a static string
+        mock_uuid = mock.patch('uuid.uuid4')
+        mock_uuid.return_value = '0000000000'
+
+        responses.add(
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            body=csv,
+            status=200,
+        )
+
+        sync_blocked_emails()
+
+        file_path = f"tmp/suppressions-{mock_uuid.return_value}.csv"
+
+        assert not default_storage.exists(file_path)
+
+        assert EmailBlock.objects.count() == 0
+
+
+
+
+
+"""
+Test scenarios:
+2. recieve bad response from api
+3. cannot reach API, timeout
+5. CSV is empty
+6. CSV contains existing email
+7. CSV contains unique email
+10. Deletes file if error occurs while processing recores.
+
+
+1. environment variables are undefined.
```

Comment:
These are test scenarios I could think of but not sure if we actually want to test. Some would be bleeding into the territory of testing the contract of socketlabs, making sure we can handle breaches in that contract, others are testing things like setttings/env variables.. not sure it's worth it. Finally, we will likely run this task with large CSV files and I didn't implement any kind of test for handling this. But since the code is batched, it should be relatively performant.